docs/modules/ROOT/pages/documentation/extensions/grpc/README.adoc

```diff
@@ -0,0 +1,401 @@
+        </dependency>
+        <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-protobuf</artifactId>
+            <version>1.43.1</version>
+        </dependency>
+        <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-stub</artifactId>
+            <version>1.43.1</version>
+        </dependency>
+        <dependency> 
+            <groupId>org.apache.tomcat</groupId>
+            <artifactId>annotations-api</artifactId>
+            <version>6.0.53</version>
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.platform</groupId>
+            <artifactId>jakarta.jakartaee-api</artifactId>
+            <version>8.0.0</version>
+            <scope>provided</scope>
+        </dependency>
+----
+
+[[build-section]]
+==== build section
+
+Example:
+[source, XML]
+----
+    <build>
+      <extensions>
+          <extension>
+              <groupId>kr.motd.maven</groupId>
+              <artifactId>os-maven-plugin</artifactId>
+              <version>1.6.2</version>
+          </extension>
+      </extensions>
+      <plugins>
+          <plugin>
+              <groupId>org.xolstice.maven.plugins</groupId>
+              <artifactId>protobuf-maven-plugin</artifactId>
+              <version>0.6.1</version>
+              <configuration>
+                  <protocArtifact>com.google.protobuf:protoc:3.19.2:exe:${os.detected.classifier}</protocArtifact>
+                  <pluginId>grpc-java</pluginId>
+                  <pluginArtifact>io.grpc:protoc-gen-grpc-java:1.43.1:exe:${os.detected.classifier}</pluginArtifact>
+              </configuration>
+              <executions>
+                  <execution>
+                      <goals>
+                          <goal>compile</goal>
+                          <goal>compile-custom</goal>
+                      </goals>
+                  </execution>
+              </executions>
+          </plugin>
+      </plugins>
+  </build>
+----
+
+[[service-implementation]]
+=== Service Implementation
+
+After doing the previous step, you can build the project to generate the stubs. 
+
+Example:
+[source, shell]
+----
+${webapp root folder} > mvn clean install
+----
+
+The stubs should be generated with similar structure like the following:
+
+* target
+** generated-sources
+*** protobuf
+**** grpc-java
+***** fish.payara.samples.grpc (*here you can see the service class*)
+**** java
+***** fish.payara.samples.grpc (*here you can see the types associated to the service*)
+
+Then you must create a class to extend the internal generated class with ImplBase particle. In our case, this class is located into PayaraServiceGrpc under the folder `/target/generated-sources/protobuf/grpc-java` .
+
+[[service-code-implementation]]
+==== Service Code Implementation
+
+[source, Java]
+----
+@Dependent
+public class PayaraService extends PayaraServiceGrpc.PayaraServiceImplBase {
```

Comment:
What should users do inside the `communicate` method? What is the contract for properly implementing the service? Any additional points that they should have into account? This part is the most important one for them as service implementors and lacks a lot of details.